## Chi-Square for Muffins 

The owner of a bakery wants to know whether customers have any preferences among 5 different flavors of muffin. A sample of 200 customers provided the data in the following list:



In [1]:
muffinList = [30,52,38,36,44]

To calculate the Chi-square statistic, we use the following formula:

$$ \chi ^2 = \sum{\frac{(o-e)^2}{e}} $$ 

o represents the observed value and e represents the expected population mean.

To calculate e, we sum the observations from each category of muffin and then divide by the number of categories.

$$ e = \frac{30 + 52 + 38 + 36 + 44}{5} $$ 

Simplified: 

$$ e = \frac{200}{5} $$ 

In [3]:
expectedMean = sum(muffinList) / len(muffinList)
print(expectedMean)

40.0


The next step is to take each observed value from each category, o, and then subtract it from e. We then square the differences and then take the sum of these squared differences. 

$$ \chi ^2 = \sum{\frac{(o-40)^2}{40}} $$ 

Category Orange:

$$ \frac{(30-40)^2}{40} = \frac{100}{40} $$ 

Category Blueberry:

$$ \frac{(52-40)^2}{40} = \frac{144}{40} $$ 

Category Chocolate:

$$ \frac{(38-40)^2}{40} = \frac{4}{40} $$ 

Category Vanilla:

$$ \frac{(36-40)^2}{40} = \frac{16}{40} $$

Category Vanilla and Raisins:

$$ \frac{(44-40)^2}{40} = \frac{16}{40} $$ 

In [5]:
for i in muffinList:
    print((i - expectedMean)**2) 

100.0
144.0
4.0
16.0
16.0


In [7]:
sqDifferences = [100,144,4,16,16] 
print(sum(sqDifferences))

280


The next step is to take the sum of these squared differences: 

$$ \chi ^2 = \frac{100}{40} + \frac{144}{40} + \frac{4}{40} + \frac{16}{40} + \frac{16}{40} = \frac{280}{40} $$ 